src/borg/archiver/diff_cmd.py

```diff
@@ -43,17 +43,36 @@ def print_text_output(diff, path):
 
         matcher = build_matcher(args.patterns, args.paths)
 
-        diffs = Archive.compare_archives_iter(
-            archive1, archive2, matcher, can_compare_chunk_ids=can_compare_chunk_ids, content_only=args.content_only
+        diffs_iter = Archive.compare_archives_iter(
+            archive1, archive2, matcher, can_compare_chunk_ids=can_compare_chunk_ids
         )
         # Conversion to string and filtering for diff.equal to save memory if sorting
-        diffs = ((path, diff.changes()) for path, diff in diffs if not diff.equal)
+        diffs_list = [diff for diff in diffs_iter if not diff.equal(args.content_only)]
 
         if args.sort:
-            diffs = sorted(diffs)
-
-        for path, diff in diffs:
-            print_output(diff, path)
+            diffs_list.sort(key=lambda diff: diff.path)
```

Comment:
hmm, guess if you kept `sorted()` here, it would work with an iterator as argument.

of course it needs to fully iterate it then and make a list before sorting, but that would only be needed if `args.sort` is True.

Reply:
still todo.